sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastExchangeExec.scala

```diff
@@ -537,9 +539,45 @@ object GpuBroadcastExchangeExecBase {
       SQLConf.get.getConf(StaticSQLConf.BROADCAST_EXCHANGE_MAX_THREAD_THRESHOLD)))
 }
 
-case class GpuBroadcastExchangeExec(mode: BroadcastMode, child: SparkPlan)
+case class GpuBroadcastExchangeExec(
+    mode: BroadcastMode,
+    child: SparkPlan)
+    (val cpuCanonical: BroadcastExchangeExec)
     extends GpuBroadcastExchangeExecBase(mode, child) {
+
+  override def otherCopyArgs: Seq[AnyRef] = Seq(cpuCanonical)
+
+  private var _isGpuPlanningComplete = false
+
+  def isGpuPlanningComplete: Boolean = _isGpuPlanningComplete
+
+  def markGpuPlanningComplete(): Unit = {
```

Comment:
Would some docs here help? This part of the code is not totally clear as to why this is being used/done.

Reply:
Updated to add some comments here.